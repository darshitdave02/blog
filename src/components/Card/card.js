import * as React from 'react'
import  HeartRed from '../../Icons/heart-red.svg'
import  HeartBlack from '../../Icons/heart-black.svg'
import {ReactComponent as Clapping} from '../../Icons/clapping.svg'


export default function Card({item}) {

    const [clapCount,setClapCount] = React.useState(0);

    const [like,setLike] = React.useState(false);

    const increaseClap = () => {
        setClapCount(clapCount+1);
    }

    const changeLike = () => {
        setLike(!like);
    } 

    return (
        <div>
          {/* {data.map((item) => ( */}
            <div className="card">

            <img src={require(`../../Images/${item.image}`)} alt={item.image} />

            <div className="details">
                <div className="meta">
                    <p>
                        {item.date}
                    </p>
                    <p>
                        {item.readingTime}
                    </p>
                </div>
                <div className="caption">
                    <p>
                        {item.title}
                    </p>
                </div>
                <div className="description">
                    <p>
                        {item.description}
                    </p>
                </div>
                <div className="interaction">
                    <div className="claps">
                        <Clapping data-testid="clapIcon" className='Clapping' onClick={increaseClap}/>
                        <div data-testid="clap">
                        {clapCount}
                        </div>
                        
                    </div>
                    <div className="likes">
                        {/* {like && <HeartRed data-testid="like" onClick={changeLike} className='HeartRed' />}
                        {!like && <HeartBlack data-testid="like" onClick={changeLike} className='HeartBlack' />} */}

                        <img src={like ? HeartRed : HeartBlack} data-testid="like" onClick={changeLike} className='HeartRed HeartBlack' alt="heartIcon" />

                    </div>

                </div>

            </div>
        </div>
        </div>
      );
}