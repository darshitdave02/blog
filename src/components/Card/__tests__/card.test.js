import Card from "../card";
import { fireEvent, render, screen } from "@testing-library/react";
import React from 'react'
import data from '../../../mockData/index.json'


describe("Card", () => {
    it("should render correctly", () => {
        const { asFragment } = render(<Card item={data[0]}/>);
        expect(asFragment()).toMatchSnapshot();
    });
    it('should increase clap count', () => {
        render(<Card item={data[0]}/>);
        const clap = screen.queryByTestId('clap');
        const clapIcon = screen.queryByTestId('clapIcon');
        expect(clap).toHaveTextContent('0');
        fireEvent.click(clapIcon);
        expect(clap).toHaveTextContent('1');
    })
    it('should change like', () => {
        render(<Card item={data[0]}/>);
        expect(screen.queryByAltText('heartIcon').src).toEqual('http://localhost/heart-black.svg');
    fireEvent.click(screen.queryByAltText('heartIcon'));
    expect(screen.queryByAltText('heartIcon').src).toEqual('http://localhost/heart-red.svg');
    })
});
