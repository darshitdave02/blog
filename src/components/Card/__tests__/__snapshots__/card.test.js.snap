// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="card"
    >
      <img
        alt="abstract.png"
        src="abstract.png"
      />
      <div
        class="details"
      >
        <div
          class="meta"
        >
          <p>
            2nd Januray, 2018
          </p>
          <p>
            2 mins
          </p>
        </div>
        <div
          class="caption"
        >
          <p>
            The future of abstract art and the culture ...
          </p>
        </div>
        <div
          class="description"
        >
          <p>
            Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
          </p>
        </div>
        <div
          class="interaction"
        >
          <div
            class="claps"
          >
            <svg
              class="Clapping"
              data-testid="clapIcon"
            >
              clapping.svg
            </svg>
            <div
              data-testid="clap"
            >
              0
            </div>
          </div>
          <div
            class="likes"
          >
            <img
              alt="heartIcon"
              class="HeartRed HeartBlack"
              data-testid="like"
              src="heart-black.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
